server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java

```diff
@@ -1294,51 +1296,7 @@ public void testValidateDataStreamsAllowsNamesThatStartsWithPrefix() {
             .build();
         // don't expect any exception when validating against (potentially backing) indices that can't create conflict because of
         // additional text before number
-        validateDataStreams(metadata.getIndicesLookup(), (DataStreamMetadata) metadata.customs().get(DataStreamMetadata.TYPE));
-    }
-
-    public void testValidateDataStreamsAllowsPrefixedBackingIndices() {
```

Comment:
This test was testing whether certain backing index naming was allowed.
Only data stream naming convention no longer cares about this.
If during a rollover a backing index by accident (which would be very rare) collides with an existing index then 
we have logic that deals with this by increasing generation (which would generate another backing index name).

@danhermann Do you think that this (^) assessment is correct? In other words this test is obsolete and can be removed?

Reply:
Yes, I think that this test has been superseded by DataStreamTests::testRolloverWithConflictingBackingIndexName.